src/storm/mesos/MesosNimbus.java

```diff
@@ -615,6 +606,57 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
       }
     }
   }
+  
+  private ExecutorInfo createExecutorInfo(Optional<ContainerInfo.Builder> containerInfoBuilder,
+          TopologyDetails details, String executorDataStr, double executorCpu, double executorMem, 
+          String cpuRole, String memRole, String configUri) {
+              
+        ExecutorInfo.Builder infoBuilder =  ExecutorInfo.newBuilder()
+                .setExecutorId(ExecutorID.newBuilder().setValue(details.getId()))
+                .setData(ByteString.copyFromUtf8(executorDataStr))
+                .addResources(Resource.newBuilder()
+                    .setName("cpus")
+                    .setType(Type.SCALAR)
+                    .setScalar(Scalar.newBuilder().setValue(executorCpu))
+                    .setRole(cpuRole))
+                .addResources(Resource.newBuilder()
+                    .setName("mem")
+                    .setType(Type.SCALAR)
+                    .setScalar(Scalar.newBuilder().setValue(executorMem))
+                    .setRole(memRole));
+                 
+        // If there is a builder then there was a docker image available.
+        if (containerInfoBuilder.isPresent()) {
+            infoBuilder.setCommand(CommandInfo.newBuilder().setShell(false).setUser("vagrant"));
+            infoBuilder.setContainer(containerInfoBuilder.get());
```

Comment:
The vagrant was a test on my part and not suppose to be there. When I did a
compare I missed it. I will remove it

On Mon, Jun 1, 2015 at 1:00 PM, Timothy Chen <notifications@github.com>
wrote:

> In src/storm/mesos/MesosNimbus.java
> <https://github.com/mesos/storm/pull/49#discussion_r31445907>:
>
> > +                .setData(ByteString.copyFromUtf8(executorDataStr))
> > +                .addResources(Resource.newBuilder()
> > +                    .setName("cpus")
> > +                    .setType(Type.SCALAR)
> > +                    .setScalar(Scalar.newBuilder().setValue(executorCpu))
> > +                    .setRole(cpuRole))
> > +                .addResources(Resource.newBuilder()
> > +                    .setName("mem")
> > +                    .setType(Type.SCALAR)
> > +                    .setScalar(Scalar.newBuilder().setValue(executorMem))
> > +                    .setRole(memRole));
> > +
> > +        // If there is a builder then there was a docker image available.
> > +        if (containerInfoBuilder.isPresent()) {
> > +            infoBuilder.setCommand(CommandInfo.newBuilder().setShell(false).setUser("vagrant"));
> > +            infoBuilder.setContainer(containerInfoBuilder.get());
>
> Why are we hardcoding vagrant user here? We won't be able to merge this
> with hardcoding a user name here.
> And I don't even think vagrant is the default user we should use?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/storm/pull/49/files#r31445907>.
>